### PR TITLE
Fix profiling array context for kernel callables

### PR DESCRIPTION
**Pull request by inducer**
Draft because:

- [x] Needs https://github.com/inducer/loopy/pull/398
- [x] Point `requirements.txt` for loopy back to `main`

Context: https://github.com/inducer/loopy/pull/222/ just merged. It broke some internal interfaces that mirgecom's profiling was using. This PR fixes that. If you see examples CI failures, that's what's happening.

### CODE REVIEW

File: mirgecom/profiling.py [hunk 1]
```diff
@@ -314,47 +317,50 @@ def _cache_kernel_stats(self, program: lp.kernel.LoopKernel, kwargs: dict) \
...
+                try:
+                    footprint = lp.gather_access_footprint_bytes(typed_t_unit)
+                    footprint_bytes = sum(footprint[k].eval_with_dict(domain_params)
+                        for k in footprint)
 
-            except lp.symbolic.UnableToDetermineAccessRange:
+                except lp.symbolic.UnableToDetermineAccessRange:
+                    footprint_bytes = None
+            else:
                 footprint_bytes = None
```
Comment (thomasgibson) [inline mirgecom/profiling.py:345]:
```python
if 0:
 ...
else:
 ...
```
:thinking:

File: mirgecom/profiling.py [hunk 2]
```diff
@@ -314,47 +317,50 @@ def _cache_kernel_stats(self, program: lp.kernel.LoopKernel, kwargs: dict) \
...
+                            .to_bytes().eval_and_sum(domain_params)
 
             flops = op_map.filter_by(dtype=[np.float32, np.float64]).eval_and_sum(
                 domain_params)
 
-            try:
-                footprint = lp.gather_access_footprint_bytes(kernel)
-                footprint_bytes = sum(footprint[k].eval_with_dict(domain_params)
-                    for k in footprint)
+            if 0:
```
Comment (inducer) [inline mirgecom/profiling.py:336]:
```suggestion
            # Footprint gathering is not yet available in loopy with kernel callables:
            # https://github.com/inducer/loopy/issues/399
            if 0:
```

File: mirgecom/profiling.py (hunk 1 above)
Comment (inducer) [reply mirgecom/profiling.py:347]: Fair point: e19525e

File: mirgecom/profiling.py [hunk 3]
```diff
@@ -314,47 +317,52 @@ def _cache_kernel_stats(self, program: lp.kernel.LoopKernel, kwargs: dict) \
...
+                            .to_bytes().eval_and_sum(domain_params)
 
             flops = op_map.filter_by(dtype=[np.float32, np.float64]).eval_and_sum(
                 domain_params)
 
-            try:
-                footprint = lp.gather_access_footprint_bytes(kernel)
-                footprint_bytes = sum(footprint[k].eval_with_dict(domain_params)
-                    for k in footprint)
+            # Footprint gathering is not yet available in loopy with kernel callables:
```
Comment (inducer) [inline mirgecom/profiling.py:336]:
```suggestion
            # Footprint gathering is not yet available in loopy with
            # kernel callables:
```

**Review by thomasgibson — APPROVED** [review]
Thanks!

**Review by matthiasdiener — APPROVED** [review]

**Pull request merged**